src/data/metadata/MetadataD2ApiRepository.ts

```diff
@@ -247,3 +300,43 @@ const formatStats = (stats: Stats) => ({
     ..._.omit(stats, ["created"]),
     imported: stats.created,
 });
+
+// (max_size=8000 - rest_url=1000) / (id_length=11 + comma_encoded=3)
+const maxIds = 500;
+
+const defaultOrder = { field: "id", order: "asc" } as const;
+
+interface GetListAllOptions {
+    type: ListMetadataParams["type"];
+    fields: object;
+    filter: Record<string, unknown>;
```

Comment:
```suggestion
    filter: Dictionary<FilterValueBase>,
```

Not sure why ``Dictionary<T>`` which is a ``Record<string, T>`` works but ``FilterBase`` does not.

Reply:
Do you mean `Record<string, FilterValueBase>`? Weird, should work.

Reply:
No ``Record<string, FilterValueBase>`` works too, I tend to use ``Dictionary`` however as it's more verbose about what it returns.

I meant that using ``FilterBase`` does not work (errors in the ``in`` access)

Reply:
Ah, now I understand what you mean. The problem is the full filter is:

```
export interface FilterBase {
    [property: string]: FilterValueBase | FilterValueBase[] | undefined;
}
```
So can have either `filter["id"].in` or `filter["id"][0].in` depending whether is an array or not. Now we are considering only the non-array case.